### PR TITLE
feat(ui): when drawing travel plan on map&mission panels, jumps use dashed lines

### PR DESCRIPTION
**Pull request by flowers9**
**Feature:**

## Feature Details
On the MapPanel, when the travel plan is drawn, connections that require a jump drive will be drawn with a dashed line instead of a normal line.  This also happens on the Mission panel for mission travel plans.  Normal lines are used for all standard hyperlink connections, even if you end up using a jump drive for them instead of a hyperdrive.

### Map Panel

Before:
Screenshot from 2023-07-30 22-30-43
After:
Screenshot from 2023-07-30 22-30-29

### Mission Panel

Before:
Screenshot from 2023-07-31 08-28-38
Kinda Middle (dashed travel, non-dashed mission):
Screenshot from 2023-07-31 08-23-44
After:
Screenshot from 2023-07-31 08-20-42

## Testing Done
Tried different travel plans, with ships both jump-capable and not, at varying zoom levels.  Jumped using jump drive to display mini-map.

## Performance Impact
A little slower in drawing the travel path for drawing the dashes, a little slower in the mission path for that and figuring out whether to draw solid or dash.

### CODE REVIEW

**Comment by TomGoodIdea** [comment]
Does this apply to the travel path when you select a job, too?

**Comment by flowers9** [comment]
No, it doesn't.  Do you think it should?  (I totally forgot about, btw, so thanks for the reminder.)

**Review by RisingLeaf — APPROVED** [review]
clean code

**Comment by flowers9** [comment]
Sorry about changing stuff around after you reviewed the code, RisingLeaf, but it occurred to me that it'd be cleaner to move the dashed draw into its own method in LineShader, particularly as it's being used in three places now (since I thought TomGoodIdea's idea was a good one).

File: source/LineShader.cpp [hunk 1]
```diff
@@ -132,3 +132,16 @@ void LineShader::Draw(const Point &from, const Point &to, float width, const Col
 	glBindVertexArray(0);
 	glUseProgram(0);
 }
+
+
+
+void LineShader::DrawDashed(const Point &from, const Point &to, const Point &unit, const float width,
```
Comment (RisingLeaf) [inline source/LineShader.cpp:138]: I am fine with this, but you should add the gap width and the dash length as arguments imo

Comment (flowers9) [reply source/LineShader.cpp:138]: I see your point (generalized routines being good and all), but perhaps we should leave that until someone needs it?

Comment (RisingLeaf) [reply source/LineShader.cpp:138]: I dont really agree on that point, but maybe you should wait on other peoples opinions

**Comment by flowers9** [comment]
Well, it's a lot more fun to play around with when you can easily change the various dash lengths, as well.  It's not fully generalized (it doesn't take an arbitrary array of length/space), but should be good enough, I think.

**Comment by flowers9** [comment]
Turns out it didn't work great when really zoomed out (very short paths didn't get drawn), so I modified the pattern length to fit at least two patterns in the given space.  This means it can display different dashed line depending on zoom and distance between the stars, which could potentially be confusing, but that contrasts with drawing a solid line (which is what one pattern length looks like), or no line, which would also be confusing.

Screenshot from 2023-07-31 10-17-23

**Comment by flowers9** [comment]
Actually, looking closely, it already messes with the pattern scaling by fitting in an integral number of them (most noticeable when it draws two for being just shy of space for three).  So this just pushes that a touch further, in the same interest of making it look nice.

**Comment by Koranir** [comment]
The ring shader has variable segments based on zoom as well.

**Comment by flowers9** [comment]
Found (and fixed) a slight error in the original code - when drawing the travel path, wormholes links that were within jump range would be drawn dashed, not solid.

Before:
Screenshot from 2023-07-31 14-55-47
After:
Screenshot from 2023-07-31 14-56-10

The new code in MissionPanel::Draw now check to see if the link is a worm or not, rather than if it's in jump range (and draw it dashed only if not a hyperlink or a wormhole).

**Comment by flowers9** [comment]
Why do my PRs keep hitting that error?  Does that check randomly fail sometimes?  (It looked reasonable enough when I tested it locally in debug mode.)

**Comment by mOctave** [comment]
I believe that test does fail randomly sometimes. Did you pass it earlier? If so, it's probably just broken again.

**Comment by Koranir** [comment]
Is it just me, or do the dashed lines have more transparency?

**Comment by flowers9** [comment]
They're the same.  However, they're written on top of the map, and they're
not opaque, so what's underneath makes a difference.  The non-dashed lines
have another line underneath them (the link or wormhole) which can make
them appear more solid.  Try looking at the edges of the travel/mission
lines, where they don't overlap the link (easier for the mission ones,
since they're wider).

**Comment by flowers9** [comment]
Incidentally, this should also now fix the bug introduced in #9104.

**Comment by flowers9** [comment]
I noticed that the travel path drawing and mission path drawing code were rather similar, so I've modified both a little bit to make them very similar.  In particular, I effectively reversed some of the comparisons/checking done in the mission code because it travels backwards along the path, but the checks were as if it was going forward (so, it would check that there was a hyperlink from the destination system to the system being jumped from, rather than the reverse).  Only relevant if hyperlinks aren't reflexive, but I figure it's good to be careful.  And, of course, with the jump and wromhole code added there, those are definitely not reflexive.

**Review by Hurleveur — APPROVED** [review]
I tried it, it works, even with wormholes and all and this looks good code wise as well, good work!
I really like how this looks - I never even noticed there was a dif between hyperspace and wormholes before

File: source/MissionPanel.cpp [hunk 2]
```diff
@@ -208,21 +209,32 @@ void MissionPanel::Draw()
 	// Update the tooltip timer [0-60].
 	hoverSortCount += hoverSort >= 0 ? (hoverSortCount < HOVER_TIME) : (hoverSortCount ? -1 : 0);
 
-	Color routeColor(.2f, .1f, 0.f, 0.f);
-	const System *system = selectedSystem;
-	while(distance.Days(system) > 0)
+	const Color routeColor(.2f, .1f, 0.f, 0.f);
```
Comment (Amazinite) [inline source/MissionPanel.cpp:212]: While we're here, could we move this color to the interfaces file?

**Review by Amazinite — COMMENTED** [review]
Looks good.

**Pull request merged**